src/custom/constants/mainMenu.ts

```diff
@@ -11,35 +11,56 @@ import IMAGE_PIE from 'assets/cow-swap/pie.svg'
 import IMAGE_SLICER from 'assets/cow-swap/ninja-cow.png'
 import IMAGE_GAME from 'assets/cow-swap/game.gif'
 
-export interface MAIN_MENU_TYPE {
+export enum MenuItemKind {
+  DROP_DOWN = 'DROP_DOWN',
+  EXTERNAL_LINK = 'EXTERNAL_LINK',
+  DARK_MODE_BUTTON = 'DARK_MODE_BUTTON',
+}
+
+export interface BasicMenuLink {
   title: string
-  url?: string
-  externalURL?: boolean
-  items?: {
-    sectionTitle?: string
-    links: {
-      title?: string
-      url?: string // If URL is an internal route
-      externalURL?: boolean // If URL is external
-      icon?: string // If icon uses a regular <img /> tag
-      iconSVG?: string // If icon is a <SVG> inline component
-      action?: string // Special purpose flag for non-regular links
-    }[]
-  }[]
+  url: string
+  icon?: string // If icon uses a regular <img /> tag
+  iconSVG?: string // If icon is a <SVG> inline component
 }
+export interface InternalLink extends BasicMenuLink {
+  kind?: undefined
```

Comment:
Kind `undefined`?

Reply:
`undefined | undefined` 👻

Reply:
It has some explanation :) 

To do Union Types discrimination, I need different values for the "kind".
To make it simpler, I leave the `kind = undefined` value to the internal link. They are the "normal case", so makes the link definition simpler.

For this reason `kind` is optional, so I don't need to specify its value.

However, to be able to discriminate, i need to tell that `InternalLink` is INDEED having undefined as the value of that optional field. Meaning, if you don't declare the kind, it is an `InternalLink`

I hope this makes things clearer

Reply:
I think I got it, but, given the explanation, it might be better to be explicit and have a type here rather than `undefined`

Reply:
not sure, i find it simple to use, and make it easier to mantain. 

Otherwise we would need to add the kind for all these 
![image](https://user-images.githubusercontent.com/2352112/174784151-df15b03e-9dcd-40f0-a31b-ace93ba97b47.png)